addons/web/static/tests/search/search_bar_tests.js

```diff
@@ -898,4 +898,69 @@ QUnit.module("Search", (hooks) => {
         await triggerEvent(searchInput, null, "keydown", { key: "ArrowUp" });
         assert.containsOnce(target, ".focus");
     });
+
+    QUnit.test("many2one_reference fields are supported in search view", async function (assert) {
+        const partnerModel = serverData.models.partner;
+        partnerModel.fields.res_id = {
+            string: "Resource ID",
+            type: "many2one_reference",
+            model_name_ref_fname: "res_model",
+            inverse_fname_by_model_name: { bar: "one2many_field" },
```

Comment:
are you sure you need such a "complicate" setup? I would be pleased if you could simplify it